backend/src/main/java/com/bakdata/conquery/models/events/stores/specific/DateRangeTypeCompound.java

```diff
@@ -0,0 +1,140 @@
+import lombok.ToString;
+
+/**
+ * Stores {@link CDateRange} by storing only the references of the start and end of the {@link CDateRange}.
+ * After the deserialisation, stores are instantiated as {@link DateStore}s using the references to the columns.
+ */
+@CPSType(base = ColumnStore.class, id = "DATE_RANGE_COMPOUND")
+@Getter
+@Setter
+@ToString(of = {"startColumn", "endColumn"})
+public class DateRangeTypeCompound implements DateRangeStore {
+
+
+	@NotNull
+	@NotEmpty
+	private String startColumn, endColumn;
+
+	// does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	@JsonIgnore
+	private DateStore startStore;
+
+	// does not have to be serialized because will be lazy-loaded after the deserialization of DateRangeTypeCompound
+	@JsonIgnore
+	private DateStore endStore;
+
+
+	/**
+	 * Represents the bucket from where this will be saved
+	 * This bucket will be set after the deserialization
+	 */
+	@Setter(AccessLevel.PROTECTED)
+	@JsonIgnore
+	private Bucket parent;
+
+	@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+	public DateRangeTypeCompound(String startColumn, String endColumn) {
+		setStartColumn(startColumn);
+		setEndColumn(endColumn);
+	}
+
+	@JsonIgnore
+	public DateStore getStartStore() {
+		if (getParent() == null) {
+			return null;
+		}
+		if (startStore == null) {
+			setStartStore((DateStore) getParent().getStore(getStartColumn()));
+		}
+		return startStore;
+	}
+
+	@JsonIgnore
+	public DateStore getEndStore() {
+		if (getParent() == null) {
+			return null;
+		}
+		if (endStore == null) {
+			setEndStore((DateStore) getParent().getStore(getEndColumn()));
+		}
+		return endStore;
+	}
+
+	@Override
+	public void setParent(@NonNull Bucket bucket) {
+		parent = bucket;
+	}
+
+	@Override
+	public int getLines() {
+		if (getStartStore() == null && getEndStore() == null) {
+			return 0;
+		}
+		if (getStartStore() == null || getEndStore() == null) {
+			throw new IllegalStateException(String.format("DateRangeTypeCompound[%s] has one null store.", this));
+		}
+		// they can be unaligned, if one of them is empty.
+		return Math.max(getStartStore().getLines(), getEndStore().getLines());
+
+	}
+
+	// Reported as 0 because this store does not hold data of its own, but references its neighbouring stores.
+	@Override
+	public long estimateEventBits() {
+		return 0;
+	}
+
+	@Override
+	public DateRangeTypeCompound select(int[] starts, int[] length) {
+		return new DateRangeTypeCompound(getStartColumn(), getEndColumn());
+	}
+
+	@Override
+	public void setDateRange(int event, CDateRange raw) {
+		// this has already done by the child stores, so no need to do it again
+	}
+
+	@Override
+	public void setNull(int event) {
+		// this has already been done by the child stores, so no need to do it again
+	}
+
+	@Override
+	public CDateRange getDateRange(int event) {
+		int start = Integer.MIN_VALUE;
+		int end = Integer.MAX_VALUE;
+
+		if (getStartStore().has(event)) {
```

Comment:
kannst du aus start und end store bitte variablen machen, sonst machst du die null checks potentiell zwei mal in ziemlich hottem code